libraries/chain/abi_serializer.cpp

```diff
@@ -133,8 +133,12 @@ namespace eosio { namespace chain {
 
    void abi_serializer::set_abi(abi_def abi, const yield_function_t& yield) {
       impl::abi_traverse_context ctx(yield, fc::microseconds{});
-
-      EOS_ASSERT(starts_with(abi.version, "eosio::abi/1."), unsupported_abi_version_exception, "ABI has an unsupported version");
+      auto abi_version = abi.get_version();
+      EOS_ASSERT(!!abi_version, unsupported_abi_version_exception, "Unrecognized abi version: ${v}",
+                 ("v", abi.version));
+      EOS_ASSERT(*abi_version <= current_abi_support, unsupported_abi_version_exception,
+                 "ABI has an unsupported version: ${v}, current nodeos supports up to: ${vc}",
+                 ("v", abi.version)("vc", current_abi_support.str()));
```

Comment:
I think we need some tests that verify get table rows handles this.
https://github.com/AntelopeIO/spring/blob/a0a56fbbd8abc39d267e6aaa6b299817c998d97d/plugins/chain_plugin/include/eosio/chain_plugin/chain_plugin.hpp#L688-L688
https://github.com/AntelopeIO/spring/blob/a0a56fbbd8abc39d267e6aaa6b299817c998d97d/plugins/chain_plugin/include/eosio/chain_plugin/chain_plugin.hpp#L800-L800

Reply:
handle in what sense? The ` get_table_row` in chain_api_plugin uses `CALL_WITH_400` which does handle exceptions occuring during the call.

Reply:
I think get table row should return the data un-deserialized if it can't use the ABI. Not just fail completely.

Reply:
I'm not sure, I think it might be better to fail completely, so that the user realizes that there is a problem and uses a up-to-date API provider.

Reply:
Reminder: If `json` is not requested don't call `set_abi` and avoid exception.

Reply:
https://github.com/AntelopeIO/spring/pull/1640/commits/31beefa74f4a0395c75434bd8c56a90f0ac5c7f0